doc/tutorials/ios-app.md

```diff
@@ -270,30 +277,95 @@ output will appear similar to the following:
 ```bash
 INFO: Found 1 target...
 Target //ios-app:ios-app up-to-date:
-  bazel-bin/ios-app/ios-app.ipa
-INFO: Elapsed time: 0.565s, Critical Path: 0.44s
+  bazel-out/applebin_ios-ios_sim_arm64-fastbuild-ST-4e6c2a19403f/bin/ios-app/ios-app.ipa
+INFO: Elapsed time: 0.141s, Critical Path: 0.00s
 ```
 
 ### Find the build outputs
 
 The `.ipa` file and other outputs are located in the
-`$WORKSPACE/bazel-bin/ios-app` directory.
+`$WORKSPACE/bazel-out/applebin_ios-ios_sim_arm64-fastbuild-ST-4e6c2a19403f/bin/ios-app/ios-app.ipa` directory.
+
+### Build the app in the simulator
+
+`rules_apple` supports running an app directly in the iOS Simulator.
+Replace `build` with `run` in the previous command to both build and
+run the application:
+
+```bash
+bazel run //ios-app:ios-app
+```
+
+Note: `--ios_simulator_device` and `--ios_simulator_version` control which
+version and device will be used when launching the app.
+
+### Generate an Xcode project
+
+There are a few community-provided solutions (such as [rules_xcodeproj](https://github.com/buildbuddy-io/rules_xcodeproj)
+and [Tulsi](https://tulsi.bazel.build/)) to help generating Xcode 
+projects. By doing so, you will be able to write, debug and test 
+iOS/macOS/watchOS/tvOS applications as if you were using the Xcode build system.
+
+Let's see how to do so with `rules_xcodeproj`.
 
-### Run and debug the app in the simulator
+Open the `WORKSPACE` file again and add the following:
 
-You can now run the app from Xcode using the iOS Simulator. First,
-[generate an Xcode project using Tulsi](http://tulsi.bazel.build/).
+```starlark
+http_archive(
+    name = "com_github_buildbuddy_io_rules_xcodeproj",
+    sha256 = "c76958e21b7ea48a40fe4130e57911670281f23be93965573b4e90be47d779b4",
+    strip_prefix = "rules_xcodeproj-47d9510b64878ae1893d355cbf134d6a4b6eb715",
+    url = "https://github.com/buildbuddy-io/rules_xcodeproj/archive/47d9510b64878ae1893d355cbf134d6a4b6eb715.tar.gz",
+)
 
-Then, open the project in Xcode, choose an iOS Simulator as the runtime scheme,
-and click **Run**.
+load(
+    "@com_github_buildbuddy_io_rules_xcodeproj//xcodeproj:repositories.bzl",
+    "xcodeproj_rules_dependencies",
+)
 
-Note: If you modify any project files in Xcode (for example, if you add or
-remove a file, or add or change a dependency), you must rebuild the app using
-Bazel, re-generate the Xcode project in Tulsi, and then re-open the project in
-Xcode.
+xcodeproj_rules_dependencies()
+```
+
+Add the following import at the top of the `BUILD` file:
+
+```starlark
+load(
+    "@com_github_buildbuddy_io_rules_xcodeproj//xcodeproj:xcodeproj.bzl",
+    "top_level_target",
+    "xcodeproj",
+)
+```
+
+We can now define the rule that will generate the Xcode project:
+
+```starlark
+xcodeproj(
+    name = "xcodeproj",
+    project_name = "ios-app",
+    tags = ["manual"],
+    build_mode = "bazel",
+    archived_bundles_allowed = True,
+    top_level_targets = [
+        top_level_target(":ios-app"),
+    ],
```

Comment:
When you aren't specifying `target_environments`, you don't need to use the macro. I would not use the macro here, but then show how to set `"device", "simulator"` down in the `Build the app for a device` section.

Reply:
Good idea, added.